File: check_db.py
import requests
import json
import datetime

url = "https://xycshsxqcfypgffnqmxb.supabase.co/rest/v1/gastos?select=cuenta,item,monto,tipo,categoria,fecha,metodo_pago,estado&order=fecha.desc&limit=10"
headers = {
    "apikey": "sb_publishable_RoIT8jS3qG_VYtX1t--h8A_vqlTyk3_",
    "Authorization": "Bearer sb_publishable_RoIT8jS3qG_VYtX1t--h8A_vqlTyk3_"
}

response = requests.get(url, headers=headers)
data = response.json()
print(json.dumps(data, indent=2, ensure_ascii=False))

File: lib/check_db.dart
import 'dart:io';
import 'package:flutter/widgets.dart';
import 'package:supabase_flutter/supabase_flutter.dart';
import 'app_secrets.dart';

Future<void> main() async {
  // Para que funcione el plugin de Flutter
  WidgetsFlutterBinding.ensureInitialized();

  await Supabase.initialize(
    url: AppSecrets.supabaseUrl,
    anonKey: AppSecrets.supabaseAnonKey,
  );
  
  final supabase = Supabase.instance.client;
  
  // Get all accounts from the user
  final response = await supabase
      .from('gastos')
      .select('cuenta, item, monto, tipo, categoria, fecha')
      .order('fecha', ascending: false)
      .limit(50);
      
  debugPrint('Last 50 movements:');
  for (var r in response) {
    debugPrint('${r['fecha']} | ${r['cuenta']} | ${r['item']} | ${r['categoria']} | ${r['tipo']} | ${r['monto']}');
  }
  exit(0);
}
